fedimint-api/src/module/registry.rs

```diff
@@ -19,28 +20,34 @@ impl<M> From<BTreeMap<ModuleInstanceId, M>> for ModuleRegistry<M> {
     }
 }
 
+impl<M> FromIterator<(ModuleInstanceId, M)> for ModuleRegistry<M> {
+    fn from_iter<T: IntoIterator<Item = (ModuleInstanceId, M)>>(iter: T) -> Self {
+        Self::new(iter)
+    }
+}
+
 impl<M> ModuleRegistry<M> {
-    pub fn new(decoders: impl IntoIterator<Item = (ModuleInstanceId, M)>) -> Self {
-        Self(decoders.into_iter().collect())
+    /// Create [`Self`] from an iterator of pairs
+    pub fn new(iter: impl IntoIterator<Item = (ModuleInstanceId, M)>) -> Self {
+        Self(iter.into_iter().collect())
     }
 
-    /// Return an iterator over all modules
+    /// Return an iterator over all module data
     pub fn iter_modules(&self) -> impl Iterator<Item = (ModuleInstanceId, &M)> {
         self.0.iter().map(|(id, m)| (*id, m))
     }
 
-    pub fn get_mut(&mut self, key: &ModuleInstanceId) -> Option<&mut M> {
-        self.0.get_mut(key)
+    /// Get module data by instance id
+    pub fn get(&self, id: ModuleInstanceId) -> Option<&M> {
+        self.0.get(&id)
     }
 
-    /// Return the server module belonging to the module identified by the supplied `module_key`
+    /// Return the module data belonging to the module identified by the supplied `module_id`
     ///
     /// # Panics
     /// If the module isn't in the registry
-    pub fn get(&self, module_key: ModuleInstanceId) -> &M {
-        self.0
-            .get(&module_key)
-            .expect("CIs were decoded, so the module exists")
+    pub fn get_expect(&self, id: ModuleInstanceId) -> &M {
```

Comment:
Seem a bit odd that `get` can panic and it might lead to bugs, but we don't want to add `expect` all over the code so we do need a method that will do it. `get_expect`... well... it does the job done.  :D